### PR TITLE
Native SDKs integration

### PR DESCRIPTION
**Pull request by ravitilicho**
# Changes
- Repo created from scratch
- Added iOS & Android native SDKs as dependencies
- Implementation complete for for `externalAsset` & `uploadExternalAsset`

## Android
- Bumped up Android SDK version to `1.0.3` and added transformer to handle `ExternalAsset` model object from `uploadExternalAsset` call response

## iOS
- Fixed React Native bridge errors (malformed arguments) due to direct return/export of async methods and Javascript promises
- Tweaked export signatures of React Native iOS methods in Objective-C file (`DashXReactNative.m`) to match Swift implementation counterparts (`DashX.swift`)

## 💡 Notes
- `FirebaseMessaging` integration is yet to be done at code level as needed

### CODE REVIEW

**Comment by ravitilicho** [comment]
@RabidFire branch is renamed to `feature/native-sdks-integration` locally but not pushed I just realised. Will rename branch after the development is complete if that works for you.

**Comment by ravitilicho** [comment]
@RabidFire SDK now supports `uploadExternalAsset` and `externalAsset` methods using native SDKs. 

Will need to expose 3 methods (`subscribe`, `screen` and `reset`) after iOS SDK 1.0.2 is live with the respective equivalents public.

File: android/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,11 +1,6 @@
...
 
-  <application>
-    <service android:name=".DashXMessagingService" android:exported="false">
-      <intent-filter>
-        <action android:name="com.google.firebase.MESSAGING_EVENT" />
-      </intent-filter>
-    </service>
-  </application>
 </manifest>
+
```
Comment (RabidFire) [inline android/src/main/AndroidManifest.xml]: Extra newlines?

File: android/build.gradle [hunk 2]
```diff
@@ -1,176 +1,50 @@
...
-        configuration = configurations.archives
-        repositories.mavenDeployer {
-            // Deploy to react-native-event-bridge/maven, ready to publish to npm
-            repository url: "file://${projectDir}/../android/maven"
-            configureReactNativePom pom
-        }
-    }
+    implementation "com.dashx:dashx-android:1.0.3"
 }
+
```
Comment (RabidFire) [inline android/build.gradle]: Extra newlines?

Comment (ravitilicho) [reply android/build.gradle]: Removed.

File: android/src/main/AndroidManifest.xml (hunk 1 above)
Comment (ravitilicho) [reply android/src/main/AndroidManifest.xml]: Removed

**Comment by ravitilicho** [comment]
@RabidFire just exposed method `saveStoredPreferences` that's missed with 1.0.2 release.

I have an open question regarding the `preferenceData` data type in Slack. Based on the decision, I might need to update the bridging code a bit. Basically Android SDK expects `JsonElement` for `preferenceData` instead of `HashMap` (as how iOS SDK expects).

**Pull request merged**